File: testcases/paper/Nested/Nested.py
class Nested:
    def __init__(self):
        self.record1 = SimpleRecord(1, 2)
        self.record2 = SimpleRecord(3, 4)

    def setfield1(self):
        self.record1.field1 = 12

    def setfield2(self):
        self.record1.field2 = 13

    def setrecords(self):
        self.record1 = self.record2

class SimpleRecord:
    def __init__(self, field1=0, field2=0):
        self.field1 = field1
        self.field2 = field2

File: testcases/paper/Nested/Nested.java
public class Nested {
    public class SimpleRecord {
        public int field1 = 0;
        public int field2 = 0;
    }

    SimpleRecord record1 = new SimpleRecord();
    SimpleRecord record2 = new SimpleRecord();

    public void setField1() {
        record1.field1 = 1;
    }

    public void setField2() {
        record1.field2 = 1;
    }

    public void setRecords() {
        record1 = record2;
    }
}